### ISSUE TITLE
Memory leak when calling AudioUtilities.GetAllDevices()

### ISSUE THREAD

**Issue by egeres** [AndreMiras/pycaw#64]
This simple test script eats all of my memory:

```python
from pycaw.pycaw import AudioUtilities

while True:
    audio_devices = AudioUtilities.GetAllDevices()
    del audio_devices[:]
```

I believe this is happening because `GetAllDevices()` from pycaw.py uses `AudioUtilities.CreateDevice(dev)` to instantiate the device itself, where this `dev` variable is of type `<class 'comtypes.POINTER(IMMDevice)'>`, which is never deleted. I have tried modifying `AudioUtilities.GetAllDevices()` itself to include a reference to this `dev` variable and override the deletion to remove it, but I've had no luck so far. I believe this needs to be implemented at C level, any ideas?

**Comment by evandrocoan** [comment]
I can confirm it:
![leak](https://user-images.githubusercontent.com/5332158/163577550-22c542fa-2aea-452a-8dd4-d400a68a41b1.gif)

**Comment by evandrocoan** [comment]
To bypass this memory leak and some comm error if I run `GetMasterVolumeLevelScalar` too many times, I am running this library in a subprocess: https://github.com/evandrocoan/timetray/commit/fc7ac3c40d85b3768276035bfc26d266c48d80e7
```python
import os
import sys
import tempfile

import pathlib
import subprocess
import threading
import pytest

g_setVolumeBase = """
import sys
import time

from ctypes import cast, POINTER
from comtypes import CLSCTX_ALL
from pycaw.pycaw import AudioUtilities, IAudioEndpointVolume

devices = AudioUtilities.GetSpeakers()
interface = devices.Activate(
    IAudioEndpointVolume._iid_, CLSCTX_ALL, None
)"""


class TemporaryFileContent:
    def __init__(self, content, suffix='.vbs'):
        self.file = tempfile.NamedTemporaryFile(mode='w', delete=False, suffix=suffix)
        with self.file as f:
            f.write(content)
            f.flush()

    @property
    def filename(self):
        return pathlib.Path(self.file.name)

    def __enter__(self):
        return self.filename

    def __exit__(self, type, value, traceback):
        # os.unlink(self.filename)
        pass


def runpython(text):
    with TemporaryFileContent(text) as filename:
        process = subprocess.Popen(f'python "{filename}"', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        stdout, stderr = process.communicate()
        stdout = stdout.decode("UTF-8", errors='replace').replace('\r\n', '\n').rstrip(' \n\r')
        threading.Timer( 1, os.unlink, args=(filename) )
        stderr = stderr.decode("UTF-8", errors='replace').replace('\r\n', '\n').rstrip(' \n\r')
        assert process.returncode == 0, f"process.returncode {process.returncode}, {stderr}, {stdout}."
        if stderr:
            print(stderr, file=sys.stderr)
        return stdout


def getSystemVolume():
    result = runpython(f"""
{g_setVolumeBase}
volume = cast(interface, POINTER(IAudioEndpointVolume))
print(volume.GetMasterVolumeLevelScalar())
        """
    )
    return float(result)


def test_get_system_volume():
    assert type(getSystemVolume()) == float

pytest.main([sys.argv[0], '-vvv',])
```

**Comment by evandrocoan** [comment]
I think I tried that and still got these problems, but I do not remember for sure. You could try it to confirm. I think as of now subprocess has enoght performance.

**Comment by KillerBOSS2019** [comment]
few things here. in `CreateDevice` this part ```
pk = store.GetAt(j)
value = store.GetValue(pk)
v = value.GetValue()```
The `value` is Prop object which needed to be released once it's done using. that's why theres part says `TODO` that says `PropVariantClear(byref(value))`. Another thing is that when you need to get stuff from the list generated from `getAllDevices` you should call `_dev.Release()` like this
```py
devices = getAllDevices()
for device in devices:
    # do whatever I want
    device._dev.Release() # once your done or release all the other device that you're not using.
```

**Comment by AndreMiras** [comment]
Since I've just merged killerBOSS' pull request, could someone else confirm it fixed the issue @egeres, @evandrocoan ?
The fix is merged in the develop branch, but not published in PyPI yet

**Comment by KillerBOSS2019** [comment]
This is the code I used
```py
from pycaw.pycaw import AudioUtilities

while True:
    audio_devices = AudioUtilities.GetAllDevices()
    del audio_devices[:]
```

Ran for 5 minutes without my PR
![image](https://user-images.githubusercontent.com/55416314/230677168-71ff2986-9282-41ae-a256-0042e1075e5b.png)

And This is with my PR ran for 5 minutes as well
![image](https://user-images.githubusercontent.com/55416314/230677531-a769c2fc-e58c-4221-937a-adfb6c90f090.png)

Default Ram usage the start. is 10mb. I think the reason why it still have little memory leak is because `_dev` is never released. So you will have to manually go thru the `audio_devices` and run `_dev.Release()`
